### PR TITLE
print typeName in ShardRegion logs

### PR DESCRIPTION
**Pull request by t3hnar**
From time to time we have some rather weird log records from ShardRegion, however taking into account that we have tens of different typeNames - it is really hard to figure out which one is causing them.

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 1]
```diff
@@ -484,7 +484,10 @@ private[akka] class ShardRegion(
   override def preStart(): Unit = {
     cluster.subscribe(self, classOf[MemberEvent])
     if (settings.passivateIdleEntityAfter > Duration.Zero)
-      log.info("Idle entities will be passivated after [{}]", PrettyDuration.format(settings.passivateIdleEntityAfter))
+      log.info(
+        "{}: Idle entities will be passivated after [{}]",
+        typeName,
```
Comment (helena) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Nice log enrichment.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 2]
```diff
@@ -544,7 +548,7 @@ private[akka] class ShardRegion(
     case msg if extractEntityId.isDefinedAt(msg) => deliverMessage(msg, sender())
     case unknownMsg =>
       log.warning(
-        "Message does not have an extractor defined in shard [{}] so it was ignored: {}",
+        "{}: Message does not have an extractor defined in shard so it was ignored: {}",
```
Comment (helena) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: But shall we keep with the convention `[{}]`?

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 3]
```diff
@@ -523,7 +526,8 @@ private[akka] class ShardRegion(
     if (before != after) {
       if (log.isDebugEnabled)
         log.debug(
-          "Coordinator moved from [{}] to [{}]",
+          "{}: Coordinator moved from [{}] to [{}]",
+          typeName,
           before.map(_.address).getOrElse(""),
```
Comment (helena) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Might this we better placement? I'm guessing for humans vs machines reading logs, it helps for the shard typeName to lead the line, but it offers less context at the start.
```scala
log.debug("Coordinator for Shard [{}] moved from [{}] to [{}]",  typeName, from, to)
```

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala (hunk 2 above)
Comment (t3hnar) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: imho: not, it does not added more readability and those typeNames are usually a kind of well readable constants

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala (hunk 3 above)
Comment (t3hnar) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: I tried to make all logs to look similar - with `${typeName}: ` as prefix

Comment (t3hnar) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: @helena but it's not a `Shard`, `shard` and `typename` are different things

Comment (helena) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Correct, I suggested based on the existing
```scala
Message does not have an extractor defined in shard [{}]
```

Comment (t3hnar) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: @helena I'm fine with suggestion, however I'd rather change 
```scala
log.warning("Message does not have an extractor defined in shard [{}] so it was ignored: {}", typeName, unknownMsg)
```

to 

```scala
log.warning("{}: Message does not have an extractor defined in shard, so it was ignored: {}", typeName, unknownMsg)
```
So not it is consistent. @helena what do you think ?

Comment (t3hnar) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: @helena actually I'd like to have asap, the actual order of words does not really matter much :)

So please propose the format I could re-use everywhere and I could update the pr there after.

**Review by johanandren — APPROVED** [review]
LGTM, good idea

**Review by helena — APPROVED** [review]
LGTM thanks, this will be helpful.

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Missing some formatting, please do a `akka-cluster-sharding/scalafmtAll` and commit the formatted sources.

**Comment by t3hnar** [comment]
@johanandren done

**Comment by t3hnar** [comment]
@johanandren will you trigger the build ?

**Comment by johanandren** [comment]
PLS BUILD

**Comment by t3hnar** [comment]
Thanks @johanandren

**Pull request merged**